txnsync/peer.go

```diff
@@ -500,15 +500,28 @@ func (p *Peer) addIncomingBloomFilter(round basics.Round, incomingFilter bloomFi
 	firstValidEntry := sort.Search(len(p.recentIncomingBloomFilters), func(i int) bool {
 		return p.recentIncomingBloomFilters[i].round >= currentRound.SubSaturate(1)
 	})
-	if firstValidEntry < len(p.recentIncomingBloomFilters) {
+	// clear out the existing bloom filters.
+	for i := 0; i < firstValidEntry; i++ {
+		p.recentIncomingBloomFilters[i] = incomingBloomFilter{}
```

Comment:
Is this necessary? Since these positions are inaccessible after if/else branches.

Reply:
clearing out the backing array is important - it ensures that the bloom filter memory can be reclaimed by the GC.